engine/classes/Elgg/Notifications/NotificationEventHandler.php

```diff
@@ -0,0 +1,356 @@
+			/* @var \ElggEntity $recipient */
+			$subject = elgg_extract('subject', $params, '');
+			$body = elgg_extract('body', $params, '');
+			$summary = elgg_extract('summary', $params, '');
+		} else {
+			$recipient = _elgg_services()->entityTable->get($guid, 'user');
+			/* @var \ElggUser $recipient */
+			if (!$recipient || $recipient->isBanned()) {
+				return false;
+			}
+		
+			if ($recipient->guid === $this->event->getActorGUID()) {
+				// Content creators should not be receiving subscription
+				// notifications about their own content
+				return false;
+			}
+			
+			if (!$actor || !$object) {
+				return false;
+			}
+
+			if ($object instanceof ElggEntity && !_elgg_services()->accessCollections->hasAccessToEntity($object, $recipient)) {
+				// Recipient does not have access to the notification object
+				// The access level may have changed since the event was enqueued
+				return false;
+			}
+
+			$subject = $this->getNotificationSubject($recipient);
+			$body = $this->getNotificationBody($recipient);
+			$summary = '';
+		}
+
+		$language = $recipient->getLanguage();
+		$params['event'] = $this->event;
+		$params['method'] = $method;
+		$params['sender'] = $actor;
+		$params['recipient'] = $recipient;
+		$params['language'] = $language;
+		$params['object'] = $object;
+		$params['action'] = $this->event->getAction();
+		$params['add_salutation'] = elgg_extract('add_salutation', $params, true);
+
+		$notification = new Notification($actor, $recipient, $language, $subject, $body, $summary, $params);
+
+		$notification = _elgg_services()->hooks->trigger('prepare', 'notification', $params, $notification);
+		if (!$notification instanceof Notification) {
+			throw new RuntimeException("'prepare','notification' hook must return an instance of " . Notification::class);
+		}
+
+		$type = 'notification:' . $this->event->getDescription();
+		$notification = _elgg_services()->hooks->trigger('prepare', $type, $params, $notification);
+		if (!$notification instanceof Notification) {
+			throw new RuntimeException("'prepare','{$type}' hook must return an instance of " . Notification::class);
+		}
+
+		if (elgg_extract('add_salutation', $notification->params) === true) {
+			$viewtype = elgg_view_exists('notifications/body') ? '' : 'default';
+			$notification->body = _elgg_view_under_viewtype('notifications/body', ['notification' => $notification], $viewtype);
+		}
+		
+		$notification = _elgg_services()->hooks->trigger('format', "notification:$method", [], $notification);
+		if (!$notification instanceof Notification) {
+			throw new RuntimeException("'format','notification:{$method}' hook must return an instance of " . Notification::class);
+		}
+
+		// return true to indicate the notification has been sent
+		$params = [
+			'notification' => $notification,
+			'event' => $this->event,
+		];
+
+		$result = _elgg_services()->hooks->trigger('send', "notification:$method", $params, false);
```

Comment:
`notification:{$method}` for readability

Reply:
done